### PR TITLE
Move open path between

### PR DESCRIPTION
**Pull request by swcraig**
Unless the piece is a knight, a piece requires an open path between the
source and destination cells in order to move there. This commit does
not yet address pawns moving forward into occupied cells. From the
movement_vector variable, `Move#path_between` builds an array of the
board cells between the source and the destination and ensures that
their contents are all nil.

This PR also contains a commit refactoring the variable naming in `spec/move_spec` to use algebraic notations for the cells. Commit message:

Using the algebraic chess notation will make variable naming more
accurate. This means that there is an "overhead" for possible future
devs on this project but I think moving to this domain-orientated naming
is justified. Otherwise, cell locations can be vague (for example,
`player_one_rook`, which rook are we referring to?), and getting used to
rank/file notation is not that difficult.

One thing to watch out for is `Cell` objects are still created with, and
use, 0 based indexing. Thus rank `a` is created by using rank = `0`.

### CODE REVIEW

File: lib/move.rb [hunk 1]
```diff
@@ -27,6 +29,30 @@ def movement_vector_and_distance
     @distance = [difference_in(:file).abs, difference_in(:rank).abs].max - 1
   end
 
+  def open_path_between_cells?
+    return true if @source_cell.content.is_a?(Knight)
```
Comment (swcraig) [inline lib/move.rb]: @brendandeere @jacquesporveau 
I was considering creating a "hook" method `Piece#can_jump?` which would be implemented to return false, and only `Knight` would override it to true. Would that be more appropriate than the explicit type check? In general, is creating a hook method that only a single subclass will override bad practice?

Comment (brendandeere) [inline lib/move.rb]: I like the idea of a hook method

File: lib/move.rb [hunk 2]
```diff
@@ -27,6 +29,30 @@ def movement_vector_and_distance
...
+    path_between(
+      @source_cell.file,
+      @source_cell.rank,
+      vector_directional_delta(@movement_vector[0]),
+      vector_directional_delta(@movement_vector[1])
+    ).map(&:content).all?(&:nil?)
```
Comment (brendandeere) [inline lib/move.rb]: you could avoids the duplicate look here by passing a block to `all?`

File: lib/move.rb [hunk 3]
```diff
@@ -27,6 +29,30 @@ def movement_vector_and_distance
...
+  end
+
+  def vector_directional_delta(delta_direction)
+    if delta_direction.zero?
```
Comment (brendandeere) [inline lib/move.rb]: This seems like there should be a better way of doing this

File: lib/pieces/piece.rb [hunk 4]
```diff
@@ -10,6 +10,11 @@ def notation
     self.class::NOTATION
   end
 
+  # Knights override this
```
Comment (brendandeere) [inline lib/pieces/piece.rb]: do we really need this comment?

**Pull request merged**